nodejs/eks/cluster.ts

```diff
@@ -627,6 +624,76 @@ export function createCore(
         },
     );
 
+    if (args.kubeProxyAddonOptions?.enabled ?? true) {
+        const kubeProxyVersion: pulumi.Output<string> = args.kubeProxyAddonOptions?.version
+            ? pulumi.output(args.kubeProxyAddonOptions?.version)
+            : aws.eks
+                  .getAddonVersionOutput(
+                      {
+                          addonName: "kube-proxy",
+                          kubernetesVersion: eksCluster.version,
+                          mostRecent: true, // whether to return the default version or the most recent version for the specified kubernetes version
+                      },
+                      { parent, provider },
+                  )
+                  .apply((addonVersion) => addonVersion.version);
+
+        const kubeProxyAddon = new aws.eks.Addon(
```

Comment:
Conditional resource, interesting. Good thing args.kubeProxyAddonOptions?.enabled are not outputty and cannot be unknown.